polar-core/src/rules.rs

```diff
@@ -80,6 +80,73 @@ impl Rule {
     }
 }
 
+// TODO: should this be a Set of Rules? Do we currently check for duplicate rules?
```

Comment:
Duplicate rules or duplicate rule types?

The only thing I can think of w/ a Set is losing the order of definition, so, e.g., if you have two rule types A & B defined in that order and you have a rule C that doesn't match either rule type, the error message might list the rule types as B & A instead of A & B. Not the end of the world.

Not sure if there's anything else we'd have to worry about.